romancal/outlier_detection/utils.py

```diff
@@ -0,0 +1,314 @@
+                computer = MedianComputer(input_shape, in_memory, buffer_size, dtype)
+                example_model = model
+                median_wcs = copy.deepcopy(model.meta.wcs)
+
+            weight_threshold = compute_weight_threshold(wht, maskpt)
+
+            data_copy = model.data.copy()
+            data_copy[wht < weight_threshold] = np.nan
+            computer.append(data_copy, i)
+
+            input_models.shelve(model, i, modify=True)
+            del data_copy, model
+
+    # Perform median combination on set of drizzled mosaics
+    median_data = computer.evaluate()
+
+    if save_intermediate_results:
+        _fileio.save_median(example_model, median_data, median_wcs, make_output_path)
+
+    return median_data, median_wcs
+
+
+def _flag_resampled_model_crs(
+    image,
+    median_data,
+    median_wcs,
+    snr1,
+    snr2,
+    scale1,
+    scale2,
+    backg,
+    save_intermediate_results,
+    make_output_path,
+):
+    blot = gwcs_blot(median_data, median_wcs, image.data.shape, image.meta.wcs, 1.0)
+
+    # Get background level of science data if it has not been subtracted, so it
+    # can be added into the level of the blotted data, which has been
+    # background-subtracted
+    if (
+        hasattr(image.meta, "background")
+        and image.meta.background.subtracted is False
+        and image.meta.background.level is not None
+    ):
+        backg = image.meta.background.level.value
+        log.debug(
+            f"Adding background level {image.meta.background.level} to blotted image"
+        )
+
+    cr_mask = flag_resampled_crs(
+        image.data.value, image.err.value, blot, snr1, snr2, scale1, scale2, backg
+    )
+
+    # update the dq flags in-place
+    image.dq |= cr_mask * np.uint32(pixel.DO_NOT_USE | pixel.OUTLIER)
+    log.info(f"{np.count_nonzero(cr_mask)} pixels marked as outliers")
+
+
+def _flag_model_crs(image, median_data, snr):
+    cr_mask = flag_crs(image.data.value, image.err.value, median_data, snr)
+
+    # Update dq array in-place
+    image.dq |= cr_mask * np.uint32(pixel.DO_NOT_USE | pixel.OUTLIER)
+
+    log.info(f"{np.count_nonzero(cr_mask)} pixels marked as outliers")
+
+
+def detect_outliers(
+    library,
+    weight_type,
+    pixfrac,
+    kernel,
+    fillval,
+    maskpt,
+    snr1,
+    snr2,
+    scale1,
+    scale2,
+    backg,
+    save_intermediate_results,
+    resample_data,
+    good_bits,
+    in_memory,
+    make_output_path,
+):
+    # setup ResampleData
+    # call
+    if resample_data:
+        resamp = ResampleData(
+            library,
+            single=True,
+            blendheaders=False,
+            # FIXME prior code provided weight_type when only wht_type is understood
+            # both default to 'ivm' but tests that set this to something else did
+            # not change the resampling weight type. For now, disabling it to match
+            # the previous behavior.
+            # wht_type=weight_type
```

Comment:
same question here

Reply:
I expanded the issue to include this as well: https://github.com/spacetelescope/romancal/issues/1428